File: 프로그래머스/unrated/160586. 대충 만든 자판/대충 만든 자판.java
import java.util.*;

class Solution {
    public int[] solution(String[] keymap, String[] targets) {
        
        int[] result = new int[targets.length];
        for(int i = 0; i < targets.length; i++)
        {
            String tStr = targets[i];
            for(int j = 0; j < tStr.length(); j++)
            {
                char curC = tStr.charAt(j);
                int keyCtn = Integer.MAX_VALUE;
                for(String key : keymap){
                    int idx = key.indexOf(curC);
                    
                    if(idx != -1 && keyCtn > idx)
                        keyCtn = idx+1;
                }

                if(keyCtn == Integer.MAX_VALUE) {
                    result[i] = -1;
                    break;
                }
                        
                
                result[i]+=keyCtn;
            }
        }
        return result;
    }
}

File: 프로그래머스/unrated/160586. 대충 만든 자판/대충 만든 자판.js
function solution(keymap, targets) {
    let result = [];

    for(let i = 0; i < targets.length; i++){
        const tStr = targets[i];
        let keyCtn = 0;
        for(let j = 0; j < tStr.length; j++){
            const curC = tStr[j];

            let tempCtn = Number.MAX_SAFE_INTEGER;
            for(let key in keymap){
                const idx = keymap[key].indexOf(curC);
                if(idx != -1 && idx < tempCtn)
                    tempCtn = idx + 1;
            }

            if(tempCtn == Number.MAX_SAFE_INTEGER) {
                keyCtn = -1;
                break;
            }

            keyCtn += tempCtn;
        }
        result.push(keyCtn);
    }

    return result;
}